### PR TITLE
Add Opt.Gear Model support

### PR DESCRIPTION
**Pull request by crinex**
## Overview

This PR adds initial support for the Gear causal language model architecture to llama.cpp.

It includes:

- HF-to-GGUF conversion support for `GearForCausalLM` / `GearModel`
- GGUF architecture and tensor mappings for Gear
- `LLM_ARCH_GEAR` model registration
- CPU graph implementation for Gear attention and local mixer layers
- Gear-specific metadata handling for:
  - per-layer attention vs local mixer layout
  - `head_dim`-based attention dimensions
  - global and sliding-window RoPE
  - shortconv key/value recurrent state
  - Gear RMSNorm with `(1 + weight)`
  - HF embedding scale `sqrt(hidden_size)`
- Quantization handling for Gear local-mixer depthwise conv tensors

## Additional information

Local validation was performed with a private Gear SFT checkpoint converted to GGUF.

Local HF-to-GGUF conversion command:

```bash
python3 convert_hf_to_gguf.py OptGear/Opt.Gear-1B \
  --outfile /path/to/gear.gguf  \
  --outtype bf16
```

HF vs llama.cpp token/logits parity passed on English, Korean, and chat-template prompts:

| Prompt | token_match | top1_match | top10 overlap | max_abs_diff | nmse |
|---|---:|---:|---:|---:|---:|
| `What is 2+2?` | true | true | 10/10 | 0.4336485863 | 0.0001060788 |
| `오늘 날씨는` | true | true | 10/10 | 0.1655693054 | 0.0000203349 |
| `<|BOT|>user\nHello<|EOT|>\n<|BOT|>assistant\n` | true | true | 10/10 | 0.6909611225 | 0.0008778222 |

Thresholds used for local parity validation:

- `max_abs_diff <= 1.0`
- `nmse <= 0.001`

Validation commands run locally:

```bash
git diff --check
python -m py_compile conversion/gear.py
cmake --build build --target llama-cli -j 4
cmake --build build --target llama-quantize -j 4
```

CPU smoke test:

```bash
build/bin/llama-simple -m /path/to/gear.gguf  -n 16 -ngl 0 "What is 2+2?"
```

Observed output included:

```text
<|BOS|>What is 2+2?

2+2=4
```

Tokenizer smoke test:

```bash
build/bin/llama-tokenize -m /path/to/gear.gguf --ids --log-disable -p "<|BOT|>user\nHello<|EOT|>\n<|BOT|>assistant\n"
```

Observed token IDs:

```text
[125030, 125039, 9604, 183, 7260, 125040, 183, 125039, 651, 14942, 183]
```

This matched the HF/llama.cpp parity token sequence.

Quantization was also tested locally with Q8_0, Q4_K_M, and Q4_0. Gear `mix_key_conv.weight` and `mix_value_conv.weight` are intentionally kept unquantized because they are small depthwise conv tensors.

Quantization commands used locally:

```bash
build/bin/llama-quantize /path/to/gear.gguf  /path/to/gear.gguf.q8_0.gguf Q8_0
build/bin/llama-quantize /path/to/gear.gguf /path/to/gear.q4_k_m.gguf Q4_K_M
build/bin/llama-quantize /path/to/gear.gguf /path/to/gear.q4_0.gguf Q4_0
```

Local CPU benchmark:

```bash
build/bin/llama-bench -m /path/to/model.gguf -p 32 -n 16 -t 4 -ngl 0 -r 3 -o md
```

| Model | Size | pp32 t/s | tg16 t/s |
|---|---:|---:|---:|
| BF16 | 1.57 GiB | 10.23 +/- 0.12 | 6.88 +/- 0.31 |
| Q8_0 | 853.73 MiB | 530.19 +/- 19.95 | 71.67 +/- 5.07 |
| Q4_K_M | 602.05 MiB | 221.89 +/- 1.20 | 91.27 +/- 2.78 |

Note: `llama-cli` in this branch starts an internal server path and failed in the local sandbox while acquiring a free port before inference. The CPU inference smoke test therefore used `llama-simple`, which directly exercises libllama model load and decode.

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: YES - AI assistance was used to help prepare this PR description. The submitted changes were reviewed and validated locally by the author.

### CODE REVIEW

**Comment by crinex** [comment]
@CISC @ggerganov 
Hello, Could you please check our request?
